rcon/blacklist.py

```diff
@@ -558,6 +600,23 @@ def edit_record_from_blacklist(
                 )
             )
 
+        # temp ban -> perma ban
+        if old_record["expires_at"] and not new_record["expires_at"]:
+            update_penalty_count(
+                player_id=record.player.player_id,
+                player=record.player,
+                action=PlayerActionState.TEMPBAN,
+                admin_name=new_record["admin_name"],
+            )
+        # perma ban -> temp ban
+        elif not old_record["expires_at"] and new_record["expires_at"]:
+            update_penalty_count(
+                player_id=record.player.player_id,
+                player=record.player,
+                action=PlayerActionState.PERMABAN,
+                admin_name=new_record["admin_name"],
+            )
+
```

Comment:
The ideal way to "unban" someone is by expiring the record. Would it be worth checking here if it is past the new expiration date?

If so, there's a few other cases to handle here as well, such as going from expired to a temporary ban.